keys/employees-keys/README.md

```diff
@@ -3,172 +3,117 @@
 
-1. [3mdeb Team Leader/Management: Identity confirmation](#identity-confirmation)
+1. [3mdeb Team Leader/Management: Identity
+   confirmation](#identity-confirmation)
 2. [3mdeb Team Leader/Management: Key signing](#key-signing)
 
 Finally you should configure your git and provide signed key through pull
 request:
 
 1. [Git configuration](#git-configuration)
 2. [Create Pull Request with signed key](#create-a-pull-request-with-key)
+3. [Upload to keys.opengpg.org](#upload-to-keys-openpgp-org)
 
 If you have questions, please check [FAQ](#FAQ) section. Maybe the answer is
 already there. If not, feel free to submit an issue to the repository.
 
-### How to create a new GPG key
+### Create a new GPG key
 
 The following assumes that you have `gpg` installed.  The interaction is broken
 into sections that correspond to user queries.
 
-```
-$ gpg --gen-key
-```
-
-For GPG version 2, you should use `gpg --full-generate-key`. Please note that
-depending on the version of GPG you use, your output may look different.
-
-This is what you'll see if `gpg` is configured for the first time:
-
-```
-gpg (GnuPG) 2.2.27; Copyright (C) 2021 Free Software Foundation, Inc.
-This is free software: you are free to change and redistribute it.
-There is NO WARRANTY, to the extent permitted by law.
-```
-
-Pick "RSA and RSA" key:
-
-```
-Please select what kind of key you want:
-   (1) RSA and RSA (default)
-   (2) DSA and Elgamal
-   (3) DSA (sign only)
-   (4) RSA (sign only)
-Your selection? 1
-```
-
-Use 4096 bits for key size:
+Clone repository:
 
-```
-RSA keys may be between 1024 and 4096 bits long.
-@What keysize do you want? (2048) 4096
-Requested keysize is 4096 bits
+```shell
+git clone https://github.com/3mdeb/3mdeb-secpack.git
 ```
 
-Use a 2 years expiry key and then confirm your choice:
+Change directory:
 
-```
-Please specify how long the key should be valid.
-         0 = key does not expire
-      <n>  = key expires in n days
-      <n>w = key expires in n weeks
-      <n>m = key expires in n months
-      <n>y = key expires in n years
-Key is valid for? (0) 2y
-Key expires at Fri 07 Apr 2023 11:40:33 AM CEST
-Is this correct? (y/N) y
+```shell
+cd 3mdeb-secpack/scripts
 ```
 
-Enter your real name:
+Generate key:
 
-```
-GnuPG needs to construct a user ID to identify your key.
-
-@Real name: Your Name
+```shell
+./gen-new-employee-key.sh "Your Name" "your.name@3mdeb.com"
 ```
 
-Company's email address:
+Please note your key id printed at the end:
 
 ```
-@Email address: your.name@3mdeb.com
-```
-
-Comment field is optional:
+(...)
+[ultimate] (1). Your Name (Employee Cert Key) <your.name@3mdeb.com>
 
-```
-@Comment:
+Your KEY_ID: D9E4EB63705C3897
 ```
 
-Check and confirm your details:
+### Encrypt revocation cert
 
-```
-You selected this USER-ID:
-    "Your Name <your.name@3mdeb.com>"
+Import 3mdeb Founder key:
 
-@Change (N)ame, (C)omment, (E)mail or (O)kay/(Q)uit? o
+```shell
+gpg --import key/owner-key/piotr-krol-key.asc
```

Comment:
@krystian-hebel I would expect 3mdeb-secpack to have a higher priority and trustworthiness than the public server. In the very long run, especially if 3mdeb will grow, I would expect 3mdeb to have its own keyserver as a source of truth.

I'm unsure how `gpg --search-keys` better solves the expired keys problem. Can you elaborate on that argument?

In the case of non-published keys, I know you want to introduce publication as a policy, but we have no such policy right now.  @macpijan @miczyg1 @TomaszAIR If we agree that keys publication of 3mdeb employees is mandatory, then maybe let's add that to scripts. If we all agree, the next question would be which server we should choose. I guess in all those, we can follow Qubes OS's best practices. The key question is, are all Qubes OS developer's keys are published and to which server?

Reply:
`gpg --search-keys` shows all keys for given search criteria (e.g. email) together with information such as expiration date. That being said, I don't think I ever used `gpg --import` to import expired keys, maybe it also shows warnings or errors.

As for publication of keys, I asked for it because it would help to get better CI for building Qubes OS packages. Right now I had to disable signature checks altogether (https://github.com/TrenchBoot/.github/pull/4/files#diff-42a392ac75e0d1fba3d4468c227c6f1e8776a783c23b105387a12cb0351d298dR78) because there is no way of getting GPG public key out of GitHub.

I agree that 3mdeb-secpack should have a higher priority. As for trustworthiness, it doesn't necessarily depend on us, it depends on policy of whoever wants to use those keys. I can imagine some companies would prefer existing public servers.

> If we agree that keys publication of 3mdeb employees is mandatory, then maybe let's add that to scripts.

How about making it part of CI executed when signed keys are uploaded? It should be possible with https://keys.openpgp.org/about/api, namely `POST /vks/v1/upload` and `POST /vks/v1/request-verify`. Key owners would have to just click the link in the mail they receive to verify.

Reply:
> gpg --search-keys shows all keys for given search criteria (e.g. email) together with information such as expiration date. That being said, I don't think I ever used gpg --import to import expired keys, maybe it also shows warnings or errors.

No import does not show that:

```shell
[22:14:06] user:3mdeb-secpack git:(add_proce_for_revocation) $ gpg --import keys/employees-keys/mike-banon-key-signed.asc 
gpg: key D9CE30E2848FE512: 1 signature not checked due to a missing key
gpg: key D9CE30E2848FE512: public key "Mike Banon <mike.banon@3mdeb.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: no ultimately trusted keys found
```

```shell
[22:15:13] user:3mdeb-secpack git:(add_proce_for_revocation) $ gpg --list-key mike.banon             
pub   rsa4096 2021-05-11 [SC] [expired: 2023-05-11]
      2CD82FED62A81BA21F31D675D9CE30E2848FE512
uid           [ expired] Mike Banon <mike.banon@3mdeb.com>
```

I guess that replacement makes sense.  I will do that for my key, but I know that not all 3mdeb Team Leaders and Managers have keys on the server. Fixed in 4a060c9c243a47cb3173435409d3c0444f7ac205

The problem is that 3mdeb employees really have to upload keys.

> How about making it part of CI executed when signed keys are uploaded? It should be possible with https://keys.openpgp.org/about/api, namely POST /vks/v1/upload and POST /vks/v1/request-verify. Key owners would have to just click the link in the mail they receive to verify.

This is a very good idea, and I agree with implementing it, but it is out of scope for this PR.  I even proposed it in April #75

Reply:
@krystian-hebel can we close this conversation, or you would expect additional improvements?